x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_enrollment_flyout/steps.tsx

```diff
@@ -65,7 +69,24 @@ export const AgentPolicySelectionStep = ({
         agentPolicies={regularAgentPolicies}
         withKeySelection={setSelectedAPIKeyId ? true : false}
         onKeyChange={setSelectedAPIKeyId}
-        onAgentPolicyChange={setSelectedPolicyId}
+        onAgentPolicyChange={async (policyID) => {
```

Comment:
maybe a `useCallack` for that callback here will be good?

Reply:
for the sake of time, I'll move this to a `useCallback`